game_board.h

```diff
@@ -0,0 +1,278 @@
+						break;
+					default:
+						break;
+				}
+			}
+		}
+		
+		void logic (){
+			/*
+			#
+			#Tail system
+			#
+			*/
+			int prevX = tailX[0];
+			int prevY = tailY[0];
+			int prev2X, prev2Y;
+			tailX[0] = xHead;
+			tailY[0] = yHead;
+			for(int i=0;i<eaten;i++){
+				prev2X = tailX[i];
+				prev2Y = tailY[i];
+				tailX[i] = prevX;
+				tailY[i] = prevY;
+				prevX = prev2X;
+				prevY = prev2Y;
+			}
+			
+			/*
+			#
+			#logic system (watch command metod)
+			#
+			*/
+			switch (dir){
+				case LEFT:
+					xHead--;
+					break;
+				case RIGHT:
+					xHead++;
+					break;
+				case UP:
+					yHead--;
+					break;
+				case DOWN:
+					yHead++;
+					break;
+				case GAME_OVER:
+					gameover=true;
+				default:
+					break;
+				}
+			/*
+			#
+			# Pacman effect || not Pacman effect
+			#
+			*/
+			if(pacmanEffect){
+				if(yHead>=height){//Funziona
+					yHead = 1;
+				}
+				else if(xHead>=width){//Funziona
+					xHead = 1;
+				}
+				else if(yHead<=0){//Funziona
+					yHead = height-1;
+				}
+				else if(xHead<=0){//funziona
+					xHead = width-1;
+				}
+			}
+			else{
+				if(yHead>=height||yHead<=0||xHead>=width||xHead<=0)
+					gameover = true;
+			}
+			
+			/*
+			#
+			# Eat Fruit effect
+			#
+			*/
+			
+			if(xHead==xFruit && yHead==yFruit){//if i eat the fruit				
+				xFruit = rand()%(height-1)+1;//I'm changing the fruit posiction
+				yFruit = rand()%(width-1)+1;//Random made
+				
+				score+=50;//You taken the fruit so you win 50 points
+				eaten+=1;//You taken the fruit so you will be more big
+			}
+			/*
+			#
+			# If you want eat your tail you will loose
+			#
+			*/
+			for(int i=0;i<eaten;i++){
+				if(tailX[i] == xHead && tailY[i] == yHead)
+					gameover = true;
+			}
+			/*
+			#
+			# If you have 1000 points you win!
+			#
+			*/
+			if(score==1000){
+				winner = true;
+				gameover=true;
+			}
+		}		
+		void draw (){
+		system ("clear");
+			cout<<"Score: "<<score<<endl;
+			for(int i=0;i<=height;i++){
+				for(int j=0;j<=width;j++){
+					if(i==0 || i==height)
+						cout<<"#";
+					//else if(j==0 || j==width)
+						//cout<<"2";
+					else if(j==0 || j==width)
+						cout<<"#";
+					else if(i==yHead && j==xHead)
+						cout<<"O";
+					else if(i==yFruit && j==xFruit)
+						cout<<"F";
+					else{
+						bool print = false;
+						for(int h=0;h<eaten;h++){
+							if(tailX[h] == j && tailY[h] == i){
+								cout<<"o";
+								print = true;
+							}
+						}
+						if(!print)
+							cout<<" ";
+					}
+				}
+			cout<<endl;				
+			}
+		}
+		
+		void finish (){
+		system("clear");
+			if(!winner){
+				system("toilet You Lose!");
+			}
+			else{
+				system("toilet You Win!");
+			}
+			string toStamp("toilet ");
+			toStamp+=nickname;
+			system(toStamp.c_str());
+			cout<<"\nAlessandro Greco <3"<<endl;
+		}
+
+		string getNickname(){return nickname;}
+		bool getGameOver () {return gameover;}
+		void setPacmanEffect (bool choise) {pacmanEffect = choise;}
+		
+		int getch(){
```

Comment:
I had to create the getch method because I created the program on linux and I couldn't use the library conio.h